src/query/server/query.go

```diff
@@ -576,7 +575,8 @@ func Run(runOpts RunOptions) {
 
 	if cfg.Carbon != nil && cfg.Carbon.Ingester != nil {
 		server, ok := startCarbonIngestion(cfg.Carbon, listenerOpts,
-			instrumentOptions, logger, m3dbClusters, downsamplerAndWriter)
+			instrumentOptions, logger, m3dbClusters, clusterNamespacesWatcher,
+			downsamplerAndWriter)
```

Comment:
nit: this is a pretty long arg list, maybe worth considering jamming it into a `CarbonSettings` or something? Happy to punt on this though since it's pretty tedious

Reply:
Yeah, I agree with you. Unfortunately, this is fairly common in the `query` startup code so probably just going to punt for now.